lib/gst/player/gstplayer.c

```diff
@@ -112,6 +122,9 @@ struct _GstPlayerPrivate
   GstPlayerState app_state;
   gint buffering;
 
+  GstPlayerMediaInfo  *media_info;  /* current media info */
+  GstPlayerMediaInfo  *media_info_copy; /* last emited media info */
```

Comment:
Here is how code works:

- If media_info object is created then *tags-changed callback update tags, caps and info in media_info object and creates a new GstPlayerMediaInfo object by copying all the contents from media_info. Finally emit the newly created object and also update media_info_copy to points  this newly created variable.
- So next time when callback is executed the same thing happens except this time we also check if media_info_copy is not NULL, if not NULL then unref the media_info_copy and update it with newly created GstPlayerMediaInfo.

So media_info_copy points the immutable GstPlayerMediaInfo object created and emited to app, and since app does not unref the object hence I needed a way to unref the object by keeping a reference in media_info_copy.

Does it make sense ?